### DISCUSSION TITLE
How to train the saved model?

### DISCUSSION THREAD

**Discussion by jabori-zy** [pycaret/pycaret#2844] [Q&A]
hello, I have been trained a model successfully and I have been saved it as .pkl file. And then I loaded it to test my data.But I find that it was not perfomed well. So I want to re-train this model. However I don't know how and I don't find the function in document.

**Comment by celestinoxp** [comment]
@ngupta23

**Reply by ngupta23** [reply]
When you say you want to retrain the model, what exactly do you mean? Do you want to retrain the same type of model with new data or keep the same data but try a different model?

In either case, you will have to do `setup` again and use any of the existing functions to create new models or tune the existing model.

**Reply by rotemgb** [reply]
@ngupta23  What do you mean by "use any of the existing functions to create new models or tune the existing model"? Does that mean that there is no way to use the saved model for retraining without running a training function, such as create_model or tune_model?

When I say retraining, I mean either retrain the same type of model with new data, and sometimes with the same data. I would want to retrain the model on the same data in case I am using a notebook, but for some reason the kernel was restarted, or the session expired.
